python/pyspark/daemon.py

```diff
@@ -102,7 +102,7 @@ def shutdown(code):
         signal.signal(SIGTERM, SIG_DFL)
         # Send SIGHUP to notify workers of shutdown
         os.kill(0, SIGHUP)
-        sys.exit(code)
+        os._exit(code)
```

Comment:
If this is really a bug in Python 3.7, let's file a bug in https://bugs.python.org/ , confirm it was a bug , and use this workaround for Python 3.7 specifically. Ideally, we should prepare a minimal reproducer and share there.

Lastly, is this test-only issue or actual problem with Python 3.7 (workers are not being dead)? If so, this issue is a critical issue that leaks resources.